Plugin/StoreLocatorBlockViewMapPlugin.php

```diff
@@ -0,0 +1,162 @@
+                            'media' => $image,
+                            'title' => $promo->getTitle(),
+                            'description' => $promo->getDescription(),
+                            'pdfFile' => $pdfFile,
+                            'link' => $promotionLink
+                        ];
+                }
+        }
+        return $result;
+    }
+
+    /**
+     * Get the list of promotions.
+     * @param int $retailerId
+     * @return PromotionInterface[]
+     * @throws \Magento\Framework\Exception\LocalizedException
+    */
+
+    public function getPromoListByRetailerId($retailerId)
+    {
+        $now = new \DateTime();
+        $currDateFormat = $now->format('Y-m-d H:i:s');
+
+        $retailerIdFilter = $this->filterBuilder
+            ->setField(PromotionInterface::RETAILER_ID)
+            ->setConditionType('in')
+            ->setValue([0, $retailerId])
+            ->create();
+        $retailerStatus = $this->filterBuilder
+            ->setField(PromotionInterface::STATUS)
+            ->setConditionType('in')
+            ->setValue([1, 2])
+            ->create();
+        $retailerActive = $this->filterBuilder
+            ->setField(PromotionInterface::IS_ACTIVE)
+            ->setValue(true)
+            ->create();
+        $retailerEndDate = $this->filterBuilder
+            ->setField(PromotionInterface::END_AT)
+            ->setConditionType('gteq')
+            ->setValue($currDateFormat)
+            ->create();
+        $retailerStartDate = $this->filterBuilder
+            ->setField(PromotionInterface::CREATED_AT)
+            ->setConditionType('lteq')
+            ->setValue($currDateFormat)
+            ->create();
+        $retailerEmptyEnd = $this->filterBuilder
+            ->setField(PromotionInterface::END_AT)
+            ->setConditionType('null')
+            ->create();
+        $retailerEmptyStart = $this->filterBuilder
+            ->setField(PromotionInterface::CREATED_AT)
+            ->setConditionType('null')
+            ->create();
+        $filterGroup = [
+            $this->filterGroupBuilder
+                ->addFilter($retailerIdFilter)
+                ->create(),
+            $this->filterGroupBuilder
+                ->addFilter($retailerStatus)
+                ->create(),
+            $this->filterGroupBuilder
+                ->addFilter($retailerActive)
+                ->create(),
+            $this->filterGroupBuilder
+                ->addFilter($retailerEndDate)
+                ->addFilter($retailerStartDate)
+                ->addFilter($retailerEmptyStart)
+                ->addFilter($retailerEmptyEnd)
+                ->create()
+        ];
+        $searchCriteria = $this->searchCriteriaBuilder
+            ->setFilterGroups($filterGroup)
+            ->create();
+        $searchResult = $this->promotionRepository->getList($searchCriteria);
+
+        $items = $searchResult->getItems();
+
+        return $items;
+    }
```

Comment:
Check with you. It's not ok

SQL will be like that 

SELECT `main_table`.*
FROM `smile_retailer_promotion` AS `main_table`
WHERE ((`retailer_id` =0 AND status = 2) OR (`retailer_id` =5 AND status = 1))
AND ((`is_active` = '1'))
AND (
 (`end_at` >= '2019-11-22 09:58:37') AND (`created_at` <= '2019-11-22 09:58:37')
 OR (`end_at` >= '2019-11-22 09:58:37' AND `created_at` = '')
 OR (`created_at` <= '2019-11-22 09:58:37' AND `end_at` = '')
 OR (`created_at` = '' AND `end_at` = '' )
)